src/corehost/cli/libhost.h

```diff
@@ -364,6 +383,14 @@ struct hostpolicy_init_t
             init->host_command = input->host_command;
         }
 
+        if (input->version_lo >= offsetof(host_interface_t, host_info_host_path) + sizeof(input->host_info_host_path))
+        {
+            init->host_info.host_path = input->host_info_host_path;
+            init->host_info.dotnet_root = input->host_info_dotnet_root;
+            init->host_info.app_path = input->host_info_app_path;
+            // For the backwards compat case, this will be later initialized with argv[0]
```

Comment:
Do we have tests for the backwards compat at all?

Reply:
Unfortunately, no. I previously created https://github.com/dotnet/core-setup/issues/3777 to track adding such tests